RecoTracker/MkFitCore/src/MkBuilder.cc

```diff
@@ -0,0 +1,1427 @@
+            "sx_h",
+            "sy_h",
+            "sz_h",
+            "x_t",
+            "y_t",
+            "z_t",
+            "r_t",
+            "sx_t",
+            "sy_t",
+            "sz_t",
+            "pt",
+            "phi",
+            "theta",
+            "phi_h",
+            "phi_t",
+            "d_xy",
+            "d_z");
+        goto redo_fit;
+      }
+#endif
+
+      // copy out full set of info at last propagated position
+      mkfndr->bkFitOutputTracks(m_tracks, icand, end, prop_config.backward_fit_to_pca);
+
+#ifdef DEBUG
+      printf("Post Final fit for %d - %d\n", icand, end);
+      for (int i = icand; i < end; ++i) {
+        const TrackCand &t = eoccs[i][0];
+        printf(
+            "  %4d with q=%+d chi2=%7.3f pT=%7.3f eta=% 7.3f x=%.3f y=%.3f z=%.3f nHits=%2d  label=%4d findable=%d\n",
+            i,
+            t.charge(),
+            t.chi2(),
+            t.pT(),
+            t.momEta(),
+            t.x(),
+            t.y(),
+            t.z(),
+            t.nFoundHits(),
+            t.label(),
+            t.isFindable());
+      }
+#endif
+    }
+  }
+
+  //------------------------------------------------------------------------------
+
+  void MkBuilder::backwardFit() {
+    EventOfCombCandidates &eoccs = m_event_of_comb_cands;
+
+    tbb::parallel_for_each(m_job->regions_begin(), m_job->regions_end(), [&](int region) {
+      const RegionOfSeedIndices rosi(m_seedEtaSeparators, region);
+
+      // adaptive seeds per task based on the total estimated amount of work to divide among all threads
+      const int adaptiveSPT = std::clamp(
+          Config::numThreadsEvents * eoccs.size() / Config::numThreadsFinder + 1, 4, Config::numSeedsPerTask);
+      dprint("adaptiveSPT " << adaptiveSPT << " fill " << rosi.count() << "/" << eoccs.size() << " region " << region);
+
+      tbb::parallel_for(rosi.tbb_blk_rng_std(adaptiveSPT), [&](const tbb::blocked_range<int> &cands) {
+        auto mkfndr = g_exe_ctx.m_finders.makeOrGet();
+
+        fit_cands(mkfndr.get(), cands.begin(), cands.end(), region);
+      });
+    });
+  }
+
+  void MkBuilder::fit_cands(MkFinder *mkfndr, int start_cand, int end_cand, int region) {
+    EventOfCombCandidates &eoccs = m_event_of_comb_cands;
+    const SteeringParams &st_par = m_job->steering_params(region);
+    const PropagationConfig &prop_config = PropagationConfig::get_default();
+
+    int step = NN;
+
+    for (int icand = start_cand; icand < end_cand; icand += step) {
+      int end = std::min(icand + NN, end_cand);
+
+      // Check if we need to fragment this for SlurpIn to work.
+      // Would actually prefer to do memory allocator for HoTNode storage.
+      // This has been "fixed" by copying Cands back into original container, not swapping the contents
+      // with vectors created in another thread (and thus not in the same memory pool, apparently), see
+      // CandCloner::processSeedRange(). Standard building does not have this problem.
```

Comment:
Could this block be removed?

Reply:
Addressed in Round 4 follow-up PR trackreco/cmssw#79 commit: https://github.com/trackreco/cmssw/pull/79/commits/d081786b240bad6e7d367004a8b05c8de61ad18b